### PR TITLE
machineconfiguration/v1alpha1: add PinnedImageSetStatus

### PR DESCRIPTION
**Pull request by hexfusion**
This PR adds a basic status to PinnedImageSet. The reason for the status which would be unused would be to ensure a Generation is populated if a set is mutated.

### CODE REVIEW

File: machineconfiguration/v1alpha1/types_pinnedimageset.go [hunk 1]
```diff
@@ -26,6 +26,20 @@ type PinnedImageSet struct {
 	// spec describes the configuration of this pinned image set.
 	// +kubebuilder:validation:Required
 	Spec PinnedImageSetSpec `json:"spec"`
+
+	// status describes the last observed state of this pinned image set.
+	// +optional
+	Status PinnedImageSetStatus `json:"status"`
```
Comment (deads2k) [inline machineconfiguration/v1alpha1/types_pinnedimageset.go:32]: missing status subresource in `// +` tags above?

**Comment by deads2k** [comment]
neat that this didn't fail on e2e tests, but there's an e2e test that ensures schemas with status have status subresources.

File: machineconfiguration/v1alpha1/types_pinnedimageset.go (hunk 1 above)
Comment (hexfusion) [reply machineconfiguration/v1alpha1/types_pinnedimageset.go:32]: gotcha kubebuilder adding

```
// +kubebuilder:subresource:status
```

**Pull request merged**